daisy/step_wait_for_instances_signal.go

```diff
@@ -224,9 +224,8 @@ func (w *WaitForInstancesSignal) run(ctx context.Context, s *Step) DError {
 			}
 			if is.SerialOutput != nil {
 				go func() {
-					if err := waitForSerialOutput(s, m["project"], m["zone"], m["instance"], is.SerialOutput, is.interval); err != nil {
-						e <- err
-					}
+					// send a signal to end other waiting serial ports, no matter whether an error happens
+					e <- waitForSerialOutput(s, m["project"], m["zone"], m["instance"], is.SerialOutput, is.interval)
```

Comment:
Would sending nil have an effect? Why this change?

Reply:
explained below in Andrew's reply. Sending nil wont break a channel.